### PR TITLE
Add contact-manager skill with SQLite

### PR DESCRIPTION
**Pull request by theodoraiakovaki**
# Description

I have created a contact-manager skill for Opsdroid, which serves as a clear example of how skills can integrate with an SQLite Database. This skill enables users to efficiently manage their contacts within the Opsdroid environment.

## Status
**READY**

## Type of change

- New feature (non-breaking change which adds functionality)
- Documentation (fix or adds documentation)

# How Has This Been Tested?

- Test A
- Test B

Tested with Slack Websockets configured as true in the configuration and with appropriate tokens.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/examples/contact-manager.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    surname = message.regex.group(3) 
+    email =  message.regex.group(4) 
+    c.execute("SELECT * FROM contacts WHERE phoneNumber = ?", (phoneNumber,))
+    existing_contact = c.fetchone()
+    if existing_contact:
+     await message.respond(f"Contact already exists!")
+    else:
+      c.execute("INSERT INTO contacts (phoneNumber, name, surname, email) VALUES (?, ?, ?, ?)", (phoneNumber, name, surname, email))
+      conn.commit()
+    await message.respond(f"Contact with Phone Number {phoneNumber} added to the Contact Book!")
```
Comment (jacobtomlinson) [inline docs/examples/contact-manager.md:106]: The indentation looka bit broken here. Could you take a look?

File: docs/examples/contact-manager.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+```
+We prevent duplicating one contact into the database by executing this:
+```python
```
Comment (jacobtomlinson) [inline docs/examples/contact-manager.md:109]: We might need some blank lines here between the code blocks and paragraphs.

File: docs/examples/contact-manager.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+@match_regex(r"Show contacts")     
+ async def showContacts(self, message):
+  c.execute("SELECT * FROM contacts")
+  rows = c.fetchall()
+  if len(rows) == 0:
+    await message.respond('No Contacts added, Please Use Command "Add contact: PhoneNumber, Name, Surname, Email"')
+  else: 
+     for row in rows:
+        phoneNumber, name, surname, email = row
+        await message.respond(f'Phone Number: {phoneNumber}, Name: {name}, Surname: {surname} and Email: {email}')
```
Comment (jacobtomlinson) [inline docs/examples/contact-manager.md:127]: Hmm indentation looks a bit broken here too. Could you check all the codeblocks?

**Review by jacobtomlinson — CHANGES_REQUESTED** [review]
This looks great, thanks for taking the time to write this.

I left a few comments below.

Also this page will need to be added to the index in https://github.com/opsdroid/opsdroid/blob/main/docs/examples/index.md.

File: docs/examples/contact-manager.md (hunk 1 above)
Comment (theodoraiakovaki) [reply docs/examples/contact-manager.md:106]: okay, of course thank you.

Comment (theodoraiakovaki) [reply docs/examples/contact-manager.md:106]: do you mean in line 104?

**Comment by theodoraiakovaki** [comment]
I have corrected what you suggested. If anything else, let me know! Thanks!

File: docs/examples/contact-manager.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  surname = message.regex.group(3) 
+  email =  message.regex.group(4) 
+  c.execute("SELECT * FROM contacts WHERE phoneNumber = ?", (phoneNumber,))
+  existing_contact = c.fetchone()
+  if existing_contact:
+    await message.respond(f"Contact already exists!")
+  else:
+    c.execute("INSERT INTO contacts (phoneNumber, name, surname, email) VALUES (?, ?, ?, ?)", (phoneNumber, name, surname, email))
+    conn.commit()
+  await message.respond(f"Contact with Phone Number {phoneNumber} added to the Contact Book!")
```
Comment (jacobtomlinson) [inline docs/examples/contact-manager.md:106]: I think you misunderstand, the spacing here is still incorrect. Each indentation needs to consist of four spaces, but all the code blocks here only use one space. Please can you fix this up.

**Review by jacobtomlinson — CHANGES_REQUESTED** [review]

File: docs/examples/contact-manager.md (hunk 4 above)
Comment (theodoraiakovaki) [reply docs/examples/contact-manager.md:106]: Okay, i see sorry, of course!

**Comment by theodoraiakovaki** [comment]
if anything else, let me know thank you!

**Review by jacobtomlinson — APPROVED** [review]

**Pull request merged**